src/EvoSC.Manialinks/ManialinkInteractionHandler.cs

```diff
@@ -0,0 +1,253 @@
+using EvoSC.Common.Interfaces.Middleware;
+using EvoSC.Common.Interfaces.Models;
+using EvoSC.Common.Interfaces.Services;
+using EvoSC.Common.Middleware;
+using EvoSC.Common.Remote;
+using EvoSC.Common.TextParsing;
+using EvoSC.Common.TextParsing.ValueReaders;
+using EvoSC.Common.Util;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Manialinks.Interfaces.Models;
+using EvoSC.Manialinks.Models;
+using GbxRemoteNet.Events;
+using GbxRemoteNet.Structs;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using SimpleInjector;
+using StringReader = EvoSC.Common.TextParsing.ValueReaders.StringReader;
+
+namespace EvoSC.Manialinks;
+
+public class ManialinkInteractionHandler : IManialinkInteractionHandler
+{
+    private readonly IManialinkActionManager _manialinkActionManager;
+    private readonly ILogger<ManialinkInteractionHandler> _logger;
+    private readonly IPlayerManagerService _players;
+    private readonly IControllerManager _controllers;
+    private readonly IActionPipelineManager _actionPipeline;
+    private readonly ValueReaderManager _valueReader = new();
+
+    public ManialinkInteractionHandler(IEventManager events, IManialinkActionManager manialinkActionManager,
+        ILogger<ManialinkInteractionHandler> logger, IPlayerManagerService playerManager,
+        IControllerManager controllers, IPlayerManagerService players,
+        IActionPipelineManager actionPipeline)
+    {
+        _manialinkActionManager = manialinkActionManager;
+        _logger = logger;
+        _controllers = controllers;
+        _players = players;
+        _actionPipeline = actionPipeline;
+
+        SetupValueReader(playerManager);
+        
+        events.Subscribe(s => s
+            .WithEvent(GbxRemoteEvent.ManialinkPageAnswer)
+            .WithInstance(this)
+            .WithInstanceClass<ManialinkInteractionContext>()
+            .WithHandlerMethod<ManiaLinkPageActionGbxEventArgs>(HandleManialinkPageAnswerAsync)
+            .AsAsync()
+        );
+    }
+
+    private void SetupValueReader(IPlayerManagerService playerManager)
+    {
+        _valueReader.AddReader(new BooleanReader());
+        _valueReader.AddReader(new FloatReader());
+        _valueReader.AddReader(new IntegerReader());
+        _valueReader.AddReader(new StringReader());
+        _valueReader.AddReader(new OnlinePlayerReader(playerManager));
+    }
+
+    private async Task HandleManialinkPageAnswerAsync(object? sender, ManiaLinkPageActionGbxEventArgs args)
+    {
+        try
+        {
+            var (action, path) = _manialinkActionManager.FindAction(args.Answer);
+
+            var (controller, context) = _controllers.CreateInstance(action.ControllerType);
+
+            if (context.ServiceScope.Container == null)
+            {
+                throw new InvalidOperationException("Service scope container is null.");
+            }
+            
+            var player = await GetPlayerAsync(args.Login);
```

Comment:
It might be possible that the player is null when evosc# was not running when the player connected.

Reply:
This shouldn't be an issue as EvoSC# requests a list of players from the server on bootup.